### PR TITLE
Add pre-send check and warning

### PR DESCRIPTION
**Pull request by Hans5958**
As it says, it adds a pre-send check and warning, which checks the message before sending and puts a warning if one or more keywords are detected on the feedback content.

The current form is quite preliminary, and may be improved on this PR based on necessity. Everyone that had access is welcome to edit it.

### CODE REVIEW

**Comment by Hans5958** [comment]
The two examples that is currently implemented.

image

image

**Comment by CST1229** [comment]
I think the warning should also direct users to the places where they can actually suggest or report.

File: static/assets/js/feedback.js [hunk 1]
```diff
@@ -80,13 +81,54 @@ const holdSendButton = seconds => {
...
+    overrideEl.textContent = i18n.preSendWarning.override
+    overrideEl.classList.add('mb-0')
+    statusEl.appendChild(overrideEl)
+    holdSendButton(5)
+}
+
+const preSendCheck = content => {
+    content = content.toLowerCase()
+    let warningText = false
+    const notSTWords = 'scratch team,st'.split(',')
```
Comment (Weredime) [inline static/assets/js/feedback.js:103]: regex these

Comment (RedGuy12) [reply static/assets/js/feedback.js:103]: and also use `\b`, what if i say the word "post" for whatever reason

**Comment by Hans5958** [comment]
> I think the warning should also direct users to the places where they can actually suggest or report to the ST.

@CST1229 This is the one I thinking to add, but I'm confused about directing to where specifically. Could you suggest what phrashes to add on that sentence?

File: static/assets/js/feedback.js [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+
+const punishmentWords = [
+    /\bban\b/,
+    /\bunban\b/,
+    /\b(please|pls|plz)\s*block\b/,
+    /\bkick\b/,
+    /\bpunish\b/,
+    /\bpunishment\b/,
+    /\bmute\b/,
+    /\bunmute\b/,
```
Comment (Samq64) [inline static/assets/js/feedback.js:14]: Mute and unmute shouldn't be here alone since we have a project muting addon.

Comment (Hans5958) [reply static/assets/js/feedback.js:14]: Nice catch. I will try to do a `pls|plz|please`.

**Comment by Hans5958** [comment]
Alright, folks. I will expedite this PR to be merged. Expect this to be merged in few days or so. You may still give comments about it.

The warnings can be expanded in the future, if we want to handle more cases in the future.

As of the "reporting to where" side of things, I haven't got an idea. What I'm considering is that there are multiple ways to report something. Do we direct it to report on the project/user page? Or on the "contact us" page? Or send them an e-mail? Or a forum post? Any other ideas?

**Comment by mxmou** [comment]
> As of the "reporting to where" side of things, I haven't got an idea. What I'm considering is that there are multiple ways to report something. Do we direct it to report on the project/user page? Or on the "contact us" page? Or send them an e-mail? Or a forum post? Any other ideas?

I think it makes sense to link to Contact Us - it has links to the forums and an option to contact the ST directly. The message could also mention the ability to report content on the Scratch website, possibly linking to [the wiki](https://en.scratch-wiki.info/wiki/Report).

**Comment by Hans5958** [comment]
Ah, I forgot there is the wiki. It explained quite well about the different methods. I think that's the best way, which is to just link it.

**Pull request merged**